fast-tags.cabal

```diff
@@ -73,7 +73,14 @@ library
         transformers,
         vector,
         void
+    if impl(ghc >= 9.2)
+        hs-source-dirs: src-9.2
+    else
+        if impl(ghc >= 8.10)
```

Comment:
Perhaps `Compat` module will also work with ghcs before 8.10?

Reply:
I think that's quite possible. The reason I limited it to 8.10 was becase I wasn't able to test any with any prior GHC versions and I didn't want to make any assumptions. If you are able to test with GHC < 8.10, that would be great!

Reply:
I'll try to set up CI so that testing happens automatically.

Reply:
Will this approach (using cabal to switch the src dir) work with ghci?  In the past I've used one module with CPP for this, where the whole thing is like

```
{#- LANGUAGE CPP #-}
module Compat ...
#if GHC < x
stuff
#endif 
```

That way I can still import everything in ghci.

Reply:
Oh wait I guess there is a `cabal repl` perhaps that will handle everything.  If so, then I have no objection!

Reply:
The `cabal repl` should resolve the `if`, but the point I tried to make in another comment in this PR is that since only 2 functions are really used by the `FastTags.LexerM` putting them there under an fdef seems like a simpler approach.

Reply:
I tend to avoid CPP if I can, but in this case I agree that it resulted in simpler code. I have pushed the CPP version.

Reply:
I have also rebased on latest master since we may have a clearer picture on CI without the non-deterministic errors.